### PR TITLE
add tools for admin of service edits

### PR DESCRIPTION
**Pull request by risicle**
This includes the `user` filtering of `find_audit_events` and the addition of the `revert_service` method.

### CODE REVIEW

**Comment by risicle** [comment]
```
./dmapiclient/data.py:14:5: C901 'DataAPIClient.find_audit_events' is too complex (12)
```
ha ha ha sod off.

**Comment by risicle** [comment]
rrright now coveralls is complaining, but I suspect that's just because I've reduced the number of lines which were covered so the proportion went down.

File: dmapiclient/data.py [hunk 1]
```diff
@@ -20,30 +20,26 @@ def find_audit_events(
             acknowledged=None,
             object_type=None,
             object_id=None,
+            user=None,
```
Comment (galund) [inline dmapiclient/data.py]: This change would be more backward-compatible if the new argument was added last in the list?

**Comment by galund** [comment]
Is there a ticket for this change?

File: dmapiclient/data.py (hunk 1 above)
Comment (risicle) [reply dmapiclient/data.py]: That it would, sir.

(though I would be more than a little worried if people were using >= 8 keyword arguments positionally)

Comment (samuelhwilliams) [reply dmapiclient/data.py]: Technically API breaking change and should be a major version bump. :trollface:

File: tests/test_apiclient.py [hunk 2]
```diff
@@ -591,6 +591,22 @@ def test_update_service_status(self, data_client, rmock):
...
+
+        data_client.revert_service(123, 314159, "jollypoldy@example.com")
+
+        assert tuple(req.json() for req in rmock.request_history) == (
+            {
+                "archivedServiceId": 314159,
+                "updated_by": "jollypoldy@example.com",
+            },
+        )
+
```
Comment (samuelhwilliams) [inline tests/test_apiclient.py]: The other tests seem to be prefer just checking `request_history[0]` directly, or `last_request`. I think this implementation confuses things by making you think about iterables when we know we're only interested in one call.

Comment (risicle) [reply tests/test_apiclient.py]: The other way doing it is fine as long as you also check `len(request_history) == 1`.

File: dmapiclient/data.py [hunk 3]
```diff
@@ -21,29 +21,26 @@ def find_audit_events(
             object_type=None,
             object_id=None,
             latest_first=None,
-            earliest_for_each_object=None):
+            earliest_for_each_object=None,
+            user=None,
+    ):
+
+        params = {
+            "page": page,
```
Comment (TheDoubleK) [inline dmapiclient/data.py]: The order of params here is pretty random, but also doesn't matter ¯\\_(ツ)_/¯

Comment (risicle) [reply dmapiclient/data.py]: True. Alphabetized now.

**Pull request merged**